### ISSUE TITLE
Changing the type of dtype argument in softmax, log_softmax, and softmin is possibly BC-breaking.

### ISSUE THREAD

**Issue by pearu** [pytorch/pytorch#68464]
## Issue description

https://github.com/pytorch/pytorch/pull/68336 changes the type of `dtype` argument of `softmax`, `log_softmax`, and `softmin` from `Optional[int]` to `Optional[DType]` where `DType = torch.types._dtype if typing.TYPE_CHECKING else int` to support MyPy and TorchScript.

This change may be BC-breaking but no evidence so far.
